### PR TITLE
[nrfconnect] Stop using Zephyr POSIX layer

### PR DESCRIPTION
**Pull request by Damian-Nordic**
After recent changes in Zephyr, the Zephyr POSIX layer no longer includes missing POSIX definitions from the standard library. Given that and that the Zephyr POSIX layer is very incomplete, it is no longer possible to build Matter samples using that layer.

Disable Zephyr socket layer, and use NET_SOCKETS_POSIX_NAMES which only wraps the socket functions with POSIX names. Make sure that proper headers are included when using this configuration.

### CODE REVIEW

File: config/zephyr/Kconfig [hunk 1]
```diff
@@ -22,14 +22,14 @@ menuconfig CHIP
...
 	imply CBPRINTF_LIBC_SUBSTS
-	imply POSIX_API if !ARCH_POSIX
 	imply EVENTFD if !ARCH_POSIX
 	imply REBOOT
 	imply ENTROPY_GENERATOR
 	imply NET_UDP
 	imply NET_IPV6
 	imply NET_CONFIG_NEED_IPV6
 	imply NET_SOCKETS
+	imply NET_SOCKETS_POSIX_NAMES if !ARCH_POSIX
```
[the remark this answers is not in the document]
Comment (Damian-Nordic) [reply config/zephyr/Kconfig:32]: Right.

**Comment by kkasperczyk-no** [comment]
> @Damian-Nordic I can see that you included [e5ca5d5](https://github.com/nrfconnect/sdk-connectedhomeip/pull/258/commits/e5ca5d5395a39d373027d111d2103b58e4a8f4f3) commit here, so I guess that PR: #234 can be closed after replacing PR number in manifest PR.



> @Damian-Nordic I can see that you included [e5ca5d5](https://github.com/nrfconnect/sdk-connectedhomeip/pull/258/commits/e5ca5d5395a39d373027d111d2103b58e4a8f4f3) commit here, so I guess that PR: #234 can be closed after replacing PR number in manifest PR.

No, your PR will be included and merged in the upmerge. PR from Damian likely will not go into upmerge and we will have to merge it once upmerge ends.

**Comment by ArekBalysNordic** [comment]
> > @Damian-Nordic I can see that you included [e5ca5d5](https://github.com/nrfconnect/sdk-connectedhomeip/pull/258/commits/e5ca5d5395a39d373027d111d2103b58e4a8f4f3) commit here, so I guess that PR: #234 can be closed after replacing PR number in manifest PR.
> 
> > @Damian-Nordic I can see that you included [e5ca5d5](https://github.com/nrfconnect/sdk-connectedhomeip/pull/258/commits/e5ca5d5395a39d373027d111d2103b58e4a8f4f3) commit here, so I guess that PR: #234 can be closed after replacing PR number in manifest PR.
> 
> No, your PR will be included and merged in the upmerge. PR from Damian likely will not go into upmerge and we will have to merge it once upmerge ends.

OK :)

**Review by rlubos — APPROVED** [review]
Ok, but it's not clear to me, what exactly broke the POSIX compatibility? Should some sort of regression be reported upstream?

**Comment by kkasperczyk-no** [comment]
> 

@rlubos actually the reason behind this change is that PR to Wi-Fi: https://github.com/nrfconnect/sdk-hostap/pull/75 Since using WPA_SUPP now depends on not using POSIX_API, we needed to align and also stop using it in Matter. Disabling POSIX_API from our side means that we don't have access to sockets API, so we decided to provide it by using NET_SOCKETS_POSIX_NAMES.

**Review by kkasperczyk-no — APPROVED** [review]

**Comment by kkasperczyk-no** [comment]
@Damian-Nordic I spotted a problem during local testing of Matter over Wi-Fi. The application fails after boot, because after disabling POSIX_API we changed the number of available socket descriptors. I think we should set CONFIG_POSIX_MAX_FDS=16 in our Kconfig.defaults. The value used to be set to 16 if we used POSIX_API. Otherwise the default value is 4.

**Pull request merged**